packages/@react-aria/color/src/useColorArea.ts

```diff
@@ -0,0 +1,578 @@
+          maskImage
+        };
+        break;
+      }
+      case 'blue': {
+        dir = xChannel === 'red';
+        maskImage = `linear-gradient(to ${orientation[Number(!dir)]}, transparent, #000)`;
+        background.colorAreaStyles = {
+          /* the background represents the red channel as a linear gradient from min to max,
+          with the green channel minimized, adjusted by the blue channel value. */
+          backgroundImage: `linear-gradient(to ${orientation[Number(dir)]},rgb(0,0,${zValue}),rgb(255,0,${zValue}))`
+        };
+        background.gradientStyles = {
+          /* the foreground represents the red channel as a linear gradient from min to max,
+          with the green channel maximized, adjusted by the blue channel value. */
+          backgroundImage: `linear-gradient(to ${orientation[Number(dir)]},rgb(0,255,${zValue}),rgb(255,255,${zValue}))`,
+          /* the foreground gradient is masked by a perpendicular linear gradient from black to white */
+          'WebkitMaskImage': maskImage,
+          maskImage
+        };
+        break;
+      }
+    }
+    return background;
+  };
+
+  let background = generateBackground();
+  let ariaRoleDescription = messages.getStringForLocale('twoDimensionalSlider', locale);
+  
+  let {visuallyHiddenProps} = useVisuallyHidden();
+  visuallyHiddenProps.style = mergeProps(
+    visuallyHiddenProps.style,
+    {
+      opacity: '0.0001',
+      width: '100%',
+      height: '100%',
+      pointerEvents: 'none'
+    }
+  );
+
+  return {
+    colorAreaProps: {
+      ...colorAriaLabellingProps,
+      ...colorAreaInteractions,
+      role: 'group',
+      style: {
+        position: 'relative',
+        touchAction: 'none',
+        ...background.colorAreaStyles
+      }
+    },
+    gradientProps: {
+      style: {
+        touchAction: 'none',
+        ...background.gradientStyles
+      }
+    },
+    thumbProps: {
+      ...thumbInteractions,
+      style: {
+        position: 'absolute',
+        left: `${x * 100}%`,
+        top: `${y * 100}%`,
+        transform: 'translate(0%, 0%)',
+        touchAction: 'none'
+      }
+    },
+    xInputProps: {
+      ...inputLabellingProps,
+      ...visuallyHiddenProps,
+      type: 'range',
+      min: state.value.getChannelRange(xChannel).minValue,
+      max: state.value.getChannelRange(xChannel).maxValue,
+      step: String(step),
+      'aria-roledescription': ariaRoleDescription,
+      'aria-valuetext': [
+        `${state.value.getChannelName(xChannel, locale)}: ${state.value.formatChannelValue(xChannel, locale)}`,
+        `${state.value.getChannelName(yChannel, locale)}: ${state.value.formatChannelValue(yChannel, locale)}`
+      ].join(', '),
+      title: getValueTitle(),
+      disabled: isDisabled,
+      value: state.value.getChannelValue(xChannel),
+      tabIndex: 0,
+      onChange: (e: ChangeEvent<HTMLInputElement>) => {
+        state.setXValue(parseFloat(e.target.value));
+      }
```

Comment:
I noticed that the ColorArea's with `hue` as one of their channels increment with a step of 2 even though the step that gets applied the the input is "1". Looks like it is possibly due to `setColorFromPoint`'s clamping behavior? Is this expected behavior? 

I also tried playing around with the size of the ColorArea to see if it had any effect but it instead broke movement via the arrow keys in the perpendicular direction (e.g. for `HSB xChannel="hue", yChannel="brightness"` story, applying `size={'size-5000'}` broke movement in the vertical direction when using the up/down arrow keys)

Reply:
Looks to be fixed now